helper/helper-common/src/main/java/com/linkedin/avroutil1/compatibility/CodeTransformations.java

```diff
@@ -548,6 +554,50 @@ public static String removeAvroGeneratedAnnotation(String code, AvroVersion minS
     return AVROGENERATED_ANNOTATION_PATTERN.matcher(code).replaceAll("");
   }
 
+  /**
+   * MODEL$ was introduced as a static field in the generated class, starting 1.8, which indicates the conversions
+   * applicable for logical types in a specific record. An example of generated code looks like:
+   * <pre>
+   * {@code
+   *   private static SpecificData MODEL$ = new SpecificData();
+   * static {
+   *     MODEL$.addLogicalTypeConversion(new org.apache.avro.data.TimeConversions.TimestampMillisConversion());
+   *   }
+   * }
+   * </pre>
+   * However, starting 1.9, avro uses reflection to look up this
+   * field, which will throw a {@link ReflectiveOperationException} exception for records generated from older version.
+   * This results in  performance degradation.
+   * Moreover, older avro runtime will not have classes used in these conversions.
+   *
+   * This methods avoids the exception by introducing this field in older versions of the generated record.
+   *
+   * @param code avro generated code that may or may not have the MODEL$ declaration
+   * @param minSupportedVersion lowest avro version under which the generated code should work
+   * @param maxSupportedVersion highest avro version under which the generated code should work
+   * @return code where MODEL$ exists for avro versions expecting it at runtime (&gt;= 1.9)
+   */
+  public static String pacifyModel$Delcaration(String code, AvroVersion minSupportedVersion,
+      AvroVersion maxSupportedVersion) {
+    if (MODEL_DECL_PATTERN.matcher(code).find()) {
+      if (minSupportedVersion.earlierThan(AvroVersion.AVRO_1_8)) {  // minAvro < 1.8
+        // replace MODEL$ with specificdata.get and remove any static block after that contains any
+        code = MODEL_DECL_PATTERN.matcher(code).replaceAll(Matcher.quoteReplacement(MODEL_DECL_REPLACEMENT));
```

Comment:
MODEL_DECL_PATTERN.matcher(code) can be defined once and reused

Reply:
Ah sure.